lib/bitcoin/payment_code.rb

```diff
@@ -0,0 +1,53 @@
+module Bitcoin
+  class PaymentCode < ExtKey
+    attr_accessor :x_value
+
+    def initialize
+      @prefix = '47'
+      @version = '01'
+      @features_bits = '00'
+      @sign = '02'
+      @reserve_field = '0' * 26
+    end
+
+    def self.generate_master(seed)
+      master_ext_key = super.derive(47, harden=true).derive(0, harden=true).derive(0, harden=true)
+
+      payment_code = PaymentCode.new
+      payment_code.depth = master_ext_key.depth
+      payment_code.key = master_ext_key.key
+      payment_code.x_value = master_ext_key.pub.slice(2...master_ext_key.pub.length) # x of pubkey
+      payment_code.chain_code = master_ext_key.chain_code
+      payment_code
+    end
+
+    def payment_code
+      Bitcoin::Base58.encode(row_payment_code + Bitcoin.calc_checksum(row_payment_code))
+    end
+
+    def row_payment_code
+      @prefix + @version + @features_bits + @sign + @x_value + @chain_code.unpack('H*').first + @reserve_field
+    end
+
+    def notification_address
+      ext_pubkey.derive(0).addr
+    end
+
+    def self.from_payment_code_string(paymen_code_string)
+      hex = Bitcoin::Base58.decode(paymen_code_string)
+      row_payment_code = hex[0...-8]
+      raise ArgumentError, 'invalid checksum' unless Bitcoin.calc_checksum(row_payment_code) == hex[-8..-1]
+
```

Comment:
The following checks are required in addition to the checksum.

* version is correct?(In this implementation, support only 0x01)
* whether x value is a point on secp256k1?
* sign value is supported?(0x02 or 0x03)?